### PR TITLE
Allow insecure crypto required by obsolete VPN gateways

### LINKED ISSUE #37: SSL error : "SSL: UNSAFE_LEGACY_RENEGOTIATION_DISABLED"
Hi @dlenski !

After upgrading from Ubuntu 20.04 to 22.04 I get the following SSL error when connecting to my office VPN :

```
gp-saml-gui -v --clientos=Windows zehost.domain.com
Looking for SAML auth tags in response to https://zehost.domain.com/global-protect/prelogin.esp...
usage: gp-saml-gui [-h] [--no-verify] [-C COOKIES | -K] [-g | -p] [-c CERT] [--key KEY] [-v | -q] [-x | -P | -S] [-u] [--clientos {Linux,Mac,Windows}] [-f EXTRA] server [openconnect_extra ...]
gp-saml-gui: error: SSL error: [SSL: UNSAFE_LEGACY_RENEGOTIATION_DISABLED] unsafe legacy renegotiation disabled (_ssl.c:997)
```
(same results using  "--clientos=Linux", for unknown reasons, IT is asking us to use  "--clientos=Windows")

It's been working fine for almost 2 years under Ubuntu 20.04. Works fine from a VM using Ubuntu 21.10.  Fails in a VM with Ubuntu 22.04. Any hints on how I could gather more details about this ?  Looking for ways to document the issue further to either bring info here, or on Ubuntu side.

Thanks !

### PR DESCRIPTION
**Pull request by DimitriPapadopoulos**
Hopefully fixes #37 and #42.

### CODE REVIEW

File: gp_saml_gui.py [hunk 1]
```diff
@@ -187,6 +189,12 @@ def parse_args(args = None):
 
     return p, args
 
+def create_insecure_context():
+    ctx = ssl.create_default_context(ssl.Purpose.CLIENT_AUTH)
+    ctx.options |= 1<<2  # OP_LEGACY_SERVER_CONNECT
+    ctx.set_ciphers('DEFAULT:-3DES:-RC4:!DH:!aNULL')
```
[the remark this answers is not in the document]
Comment (DimitriPapadopoulos) [reply gp_saml_gui.py:195]: Yes, it should be `+3DES:+RC4`, sorry about that.

I agree about `!DH`, yet in practice I believe it (almost?) always fixes the `dh key too small` error without breaking anythign else nowadays. See for example [adrienverge/openfortivpn#725 (comment 636346936)](https://github.com/adrienverge/openfortivpn/issues/725#issuecomment-636346936).

[the remark this answers is not in the document]
Comment (DimitriPapadopoulos) [reply gp_saml_gui.py]: In openfortivpn we have two options:
```c
	/* Disable vulnerable TLS protocols and ciphers by default*/
	if (!tunnel->config->cipher_list) {
		if (!tunnel->config->insecure_ssl) {
			const char *cipher_list;

			if (tunnel->config->seclevel_1)
				cipher_list = "HIGH:!aNULL:!kRSA:!PSK:!SRP:!MD5:!RC4@SECLEVEL=1";
			else
				cipher_list = "HIGH:!aNULL:!kRSA:!PSK:!SRP:!MD5:!RC4";
			tunnel->config->cipher_list = strdup(cipher_list);
		} else if (tunnel->config->seclevel_1) {
			static const char cipher_list[] = "DEFAULT@SECLEVEL=1";

			tunnel->config->cipher_list = strdup(cipher_list);
		}
	}
```
I suspect `DEFAULT@SECLEVEL=1` is good enough nowadays.

[the remark this answers is not in the document]
Comment (DimitriPapadopoulos) [reply gp_saml_gui.py]: The alternative could be to catch exceptions and try every single trick that could work sequentially:
- ciphers with `@SECLEVEL=1`
- ciphers with `+3DES:+RC4`
- ciphers with `!DH`
- option `SSL_OP_LEGACY_SERVER_CONNECT`
- configuration file `OPENSSL_CONF=~/ssl.conf` with any of the above workarounds if they cannot be directly applied programmatically.

[the remark this answers is not in the document]
Comment (dlenski) [reply gp_saml_gui.py]: Yeah, it'll get pretty messy pretty quickly. I think what you've arrived at (`+3DES:+RC4:!DH@SECLEVEL=1`) does seem like the most-likely-to-work compromise for now.

OpenConnect has the `--gnutls-priority`/`--openssl-ciphers` options for the most pathological cases :sob: 
https://github.com/openconnect/openconnect/blob/master/main.c#L1972-L1976

[the remark this answers is not in the document]
Comment (DimitriPapadopoulos) [reply gp_saml_gui.py]: I missed these two options, they are not documented. Should I add them to this script. Unfortunately, we don't know whether OpenConnect has been built with OpenSSL or GnuTLS, so we don't know which of these two options to use and then pass to OpenConnect.

**Pull request merged**